pkg/controllers/nodeclaim/lifecycle/launch_test.go

```diff
@@ -34,24 +35,43 @@ var _ = Describe("Launch", func() {
 	BeforeEach(func() {
 		nodePool = test.NodePool()
 	})
-	It("should launch an instance when a new NodeClaim is created", func() {
-		nodeClaim := test.NodeClaim(v1.NodeClaim{
-			ObjectMeta: metav1.ObjectMeta{
-				Labels: map[string]string{
-					v1.NodePoolLabelKey: nodePool.Name,
+	DescribeTable(
```

Comment:
Since it looks like these tables just change behavior for the not-managing cloudproviders with a single param, should this be a `Context` that sets up the behavior for the managing cloudprovider and then alters the nodeClaimOpts in the `It` block for managed?

Reply:
The idea was to eliminate false positives from changes in the launch conditions. I think the best example is initialization, where we need to meet a bunch of initial conditions to pass the test. If those conditions change, but we don't update the non-managed test to reflect that, it may continue to pass even though it shouldn't. By having a "base" test and ensuring that the only difference is if the resource is managed or not, we ensure we don't have skew between tests causing false positives.

Reply:
I do find this approach a little hacky fwiw, but I think it's worth ensuring we don't regress blindly due to test skew.